feat(webmodeler): H2, MariaDB and MySQL configuration

JacobiCamunda opened the pull request:
## Description

Closes https://github.com/camunda/web-modeler/issues/18465
Closes https://github.com/camunda/web-modeler/issues/18174

## When should this change go live?

<!-- PRs merged go to stage.docs.camunda.io first and must be manually released to docs.camunda.io. -->

- [ ] ~This is a bug fix, security concern, or something that needs **urgent release support**. (add `bug` or `support` label)~
- [ ] ~This is already available but undocumented and should be released within a week. (add `available & undocumented` label)~
- [ ] ~This is on a **specific schedule** and the assignee will coordinate a release with the Documentation team. (create draft PR and/or add `hold` label)~
- [x] This is part of a scheduled **alpha or minor**. (add alpha or minor label) (**8.9.0-alpha1**)
- [ ] ~There is **no urgency** with this change (add `low prio` label)~

## PR Checklist

- [x] The commit history of this PR is cleaned up, using [`{type}(scope): {description}` commit message(s)](https://github.com/camunda/camunda-docs/blob/main/CONTRIBUTING.MD#commit-message-header-formatting)

<!-- Camunda maintains 18 months of minor versions. Backporting your change to multiple versions is common. -->

- [x] My changes are for **an upcoming minor release** and are in the `/docs` directory (version ~8.8~ 8.9).
- [ ] ~My changes are for an **already released minor** and are in a `/versioned_docs` directory.~

<!-- UNCOMMENT THIS SECTION IF APPLICABLE. Adding or removing pages requires extra steps.
- [ ] I included my new page in the sidebar file(s).
- [ ] I added a redirect for a renamed or deleted page to the .htaccess file.
-->

- [x] I added a DRI, team, or delegate as a reviewer for technical accuracy and grammar/style:
  - [x] [Engineering team review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process)
  - [x] [Technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process) via `@camunda/tech-writers` unless working with an embedded writer.

<!-- UNCOMMENT THIS SECTION IF APPLICABLE. Changes to **docs infra**, including updates to workflows and adding new npm packages, must be first discussed via issue or #ask-c8-documentation and linked for context.
- [ ] My changes require a [docs infrastructure review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process). (add `dx` label) -->

mesellings commented:
@christinaausley I've assigned to you for review as I think this probably relates to your database docs work?

JacobiCamunda commented on docs/self-managed/components/modeler/web-modeler/configuration/database.md:
```diff
@@ -130,10 +300,10 @@ webModeler:
           ]
         volumeMounts:
-          - name: oracle-driver
+          - name: mysql-driver
```
Yeah, I also asked that Andreas already and he meant it can be removed.

JacobiCamunda commented on docs/self-managed/components/modeler/web-modeler/configuration/configuration.md:
```diff
@@ -89,8 +89,7 @@ Refer to the [Oracle database configuration section](database.md#oracle) for det
 
-Using a custom database schema via `SPRING_DATASOURCE_HIKARI_SCHEMA` is supported for PostgreSQL only. Oracle does not support custom schemas. MSSQL supports custom schemas, but this is not configurable in Web Modeler.
-If you want to use a custom schema in MSSQL, you must set the default schema for the database user accordingly.
+Using a custom database schema via `SPRING_DATASOURCE_HIKARI_SCHEMA` is supported for PostgreSQL only.
```
Yep, you are right. Removed

Szik commented on docs/self-managed/components/modeler/web-modeler/configuration/database.md:
```diff
@@ -80,16 +80,186 @@ instance, in addition to the adjustments described [above](#running-web-modeler-
+
+MSSQL is case-insensitive by default. To enable case sensitivity, set the database collation to a case-sensitive one like `Latin1_General_CS_AS`.
+
+Not doing so may lead to unexpected behavior.
```
Otherwise, you may encounter unexpected behavior.

Szik commented on docs/self-managed/components/modeler/web-modeler/configuration/database.md:
```diff
@@ -80,16 +80,186 @@ instance, in addition to the adjustments described [above](#running-web-modeler-
+#### Custom schema
+
+MSSQL supports custom schemas, but this is not configurable in Web Modeler.
+If you want to use a custom schema in MSSQL, you must set the default schema for the database user accordingly.
```
MSSQL supports custom schemas, but this isn’t configurable in Web Modeler. To use a custom schema, set the database user’s default schema.

Szik commented on docs/self-managed/components/modeler/web-modeler/configuration/database.md:
```diff
@@ -80,16 +80,186 @@ instance, in addition to the adjustments described [above](#running-web-modeler-
+
+### MySQL
+
+As the MySQL driver is not provided by default in each of the Camunda 8 distributions, you must download the driver and supply it for the application to load.
```
i would write:

"As the MySQL driver is not included by default in each of the Camunda 8 distributions, you must download the driver and provide it for the application to load."

Szik commented on docs/self-managed/components/modeler/web-modeler/configuration/database.md:
```diff
@@ -143,21 +313,32 @@ webModeler:
+MySQL usually uses case-insensitive collations by default. To enable case sensitivity, set the database collation to a case-sensitive one like `utf8mb4_0900_as_cs`.
+
+Not doing so may lead to unexpected behavior.
+The only known restriction currently is that extraction fields in [IDP extraction](../../../../../components/modeler/web-modeler/idp/idp-unstructured-extraction.md#extract-fields) will not be case-sensitive.
```
no "known" here too

Szik commented on docs/self-managed/components/modeler/web-modeler/configuration/database.md:
```diff
@@ -143,21 +313,32 @@ webModeler:
+
+### Oracle
+
+As the Oracle driver is not provided by default in each of the Camunda 8 distributions, you must download the driver and supply it for the application to load.
```
As the Oracle driver is not included by default in each of the Camunda 8 distributions, you must download the driver and provide it for the application to load.

JacobiCamunda commented:
@christinaausley, I addressed all comments from @Szik. You can give your final review. @aleksander-dytko, feel free to also add your feedback. If not, the PR is ready to merge after I then addressed @christinaausley's comments.

christinaausley reviewed (dismissed):
@JacobiCamunda Feel free to review my latest commit with all of my adjustments and let me know if you have questions 🚀

JacobiCamunda commented on docs/self-managed/components/modeler/web-modeler/configuration/database.md:
```diff
@@ -1,162 +1,207 @@
+| Database   | Default driver included | Notes                             |
+| ---------- | ----------------------- | --------------------------------- |
+| PostgreSQL | ✅ Yes                  | Default database                  |
+| H2         | ✅ Yes                  | For development or testing only   |
```
🔧 And also evaluation for H2

christinaausley commented on docs/self-managed/components/modeler/web-modeler/configuration/database.md:
```diff
@@ -166,52 +211,56 @@ webModeler:
-
-Otherwise, you may encounter unexpected behavior.
-The only restriction currently is that extraction fields in [IDP extraction](../../../../../components/modeler/web-modeler/idp/idp-unstructured-extraction.md#extract-fields) will not be case-sensitive.
-This means that if you have a field named `amount`, you can't create another field named `Amount`, because the database treats these two names as the same identifier.
```
You’re right — I’ve restored the restriction note for MariaDB so it’s consistent with the guidance for the other vendors 👍

christinaausley replied:
Done 😄

JacobiCamunda commented on docs/self-managed/components/modeler/web-modeler/configuration/database.md:
```diff
@@ -1,162 +1,207 @@
+| ---------- | ----------------------- | --------------------------------- |
+| PostgreSQL | ✅ Yes                  | Default database                  |
+| H2         | ✅ Yes                  | For development or testing only   |
+| MariaDB    | ✅ Yes                  | Case-sensitive collation optional |
```
I don't think "Supported for production use." is necessary because all vendors except H2 are supported for that. In my opinion, it could just be kept empty 🙂

christinaausley commented:
Releasing docs now @JacobiCamunda 👍
